pkg/await/core_service.go

```diff
@@ -192,9 +192,12 @@ func (sia *serviceInitAwaiter) await(
 	settled chan struct{},
 ) error {
 	inputServiceName := sia.config.currentInputs.GetName()
+
+	sia.config.log(diag.Info, "[1/3] Finding Pods to direct traffic to", true)
+
 	for {
 		// Check whether we've succeeded.
-		if sia.serviceReady && sia.endpointsSettled && sia.endpointsReady {
+		if sia.succeeded() {
```

Comment:
I think I'd prefer `succeeded` to be side-effect free and then log stuff in the `true` branch of this if. Based on the name and its usage it's easy to think that `succeeded` is side-effect free.